python/lsst/analysis/drp/scatterPlot.py

```diff
@@ -0,0 +1,532 @@
+            calculated from the data.  If being given a tuple of
+            (yMin, yMax).
+        xLims : `Bool` or `tuple`, optional
+            The x axis limits to use for the plot.  If `False`, they are
+            calculated from the data.
+            If being given a tuple of (xMin, xMax).
+
+        Returns
+        -------
+        fig : `matplotlib.figure.Figure`
+            The resulting figure.
+
+        Notes
+        -----
+        Uses the config options `xColName` and `yColName` to plot a scatter
+        plot of the values against each other. A histogram of the points
+        collapsed onto each axis is also plotted. A summary panel showing the
+        median of the y value in each patch is shown in the upper right corner
+        of the resultant plot. The code uses the column labelled `useForStats`
+        to determine which points to use for calculating the statistics.
+        """
+        self.log.info("Plotting {}: the values of {} on a scatter plot.".format(
+                      self.config.connections.plotName, self.config.axisLabels["y"]))
+
+        fig = plt.figure()
+        gs = gridspec.GridSpec(4, 4)
+
+        # Need to separate stars and galaxies
+        stars = (catPlot["sourceType"] == 1)
+        galaxies = (catPlot["sourceType"] == 2)
+
+        xCol = self.config.axisLabels["x"]
+        yCol = self.config.axisLabels["y"]
+        magCol = self.config.axisLabels["mag"]
+
+        # For galaxies
+        xsGalaxies = catPlot.loc[galaxies, xCol]
+        ysGalaxies = catPlot.loc[galaxies, yCol]
+
+        # For stars
+        xsStars = catPlot.loc[stars, xCol]
+        ysStars = catPlot.loc[stars, yCol]
+
+        highStats = []
+        lowStats = []
+        # Calculate some statistics
+        if np.any(catPlot["sourceType"] == 2):
+            highSnGals = ((catPlot["useForStats"] == 1) & galaxies)
+            highSnGalMed = np.nanmedian(catPlot.loc[highSnGals, yCol])
+            highSnGalMad = sigmaMad(catPlot.loc[highSnGals, yCol], nan_policy="omit")
+
+            lowSnGals = (((catPlot["useForStats"] == 1) | (catPlot["useForStats"] == 2)) & galaxies)
+            lowSnGalMed = np.nanmedian(catPlot.loc[lowSnGals, yCol])
+            lowSnGalMad = sigmaMad(catPlot.loc[lowSnGals, yCol], nan_policy="omit")
+
+            highStats += ["Median: {:0.2f}".format(highSnGalMed),
+                          r"$\sigma_{MAD}$: " + "{:0.2f}".format(highSnGalMad)]
+            lowStats += ["Median: {:0.2f}".format(lowSnGalMed),
+                         r"$\sigma_{MAD}$: " + "{:0.2f}".format(lowSnGalMad)]
+
+        if np.any(catPlot["sourceType"] == 1):
+            highSnStars = ((catPlot["useForStats"] == 1) & stars)
+            highSnStarMed = np.nanmedian(catPlot.loc[highSnStars, yCol])
+            highSnStarMad = sigmaMad(catPlot.loc[highSnStars, yCol], nan_policy="omit")
+            approxHighMag = np.nanmax(catPlot.loc[highSnStars, magCol])
```

Comment:
OK, I think I see the role of `magCol` now. It's to allow any generic plot with any x column, and any generic selection for stats, but then, whatever that is, to always report a mag here instead of a value of `x`?  It seems a bit specialized.  Might help with understanding its role to also have an example demo plot that shows a case where `magCol` and `xCol` are not the same, but where `magCol` would be missed if it didn't exist.

Reply:
I've added an example that covers both this and the points you made on the jira ticket.